src/content/sessions/2023-04-21.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Enhancements to generated XML Doc code-fixes
+preview: Enhancements to generated XML Doc code-fixes
+isDraft: false
+date: 2023-04-21
+slug: "2023/04/21"
+champion: David Schaefer
+company: GR-OSS
+repository: https://github.com/fsharp/FsAutoComplete/issues/1104
+---
+
+## Problem
+
+Enhancements to generated XML Doc code-fixes
+
+There are two things the command could be improved on:
+
+- Don't add xml doc lines if there are already some doc lines present.
+- Work on all possible AST elements, not just the ones for which we get a result from TryGetSignatureData.
+
+```fsharp
+testCaseAsync "documentation on member"
+   <| CodeFix.check
+     server
+     """
+     type MyClass() =
+       n$0ew(x: int) = MyClass()
+     """
+     Diagnostics.acceptAll
+     selectCodeFix
+     """
+     type MyClass() =
+       /// <summary></summary>
+       /// <param name="x"></param>
+       /// <returns></returns>
+       new(x: int) = MyClass()
+     """
+```
+
+## Expected behavior
+- Don't add xml doc lines if there are already some doc lines present.
+
+## Champion
+
+- [David Schaefer](https://twitter.com/dawe70704856)
+
+## Links
+
+- Date: Apr 21, 2023 10:00 (GMT-6:00) Mountain Daylight Time (<small>or 16:00 (GMT+2:00) Central European Summer Time</small>)
```

Comment:
Nitpick but I put the timing based on the speaker.
```suggestion
- Date: Apr 21, 2023 15:00 (GMT+2:00) Central European Summer Time (<small>or 09:00 (GMT-4:00) Eastern Time</small>)
```

And we would like to start at 3pm our time, 2pm UK time.